gen/templates/datasource_test.go.tmpl

```diff
@@ -110,7 +116,9 @@ data "aci_{{$.resourceName}}" "test_non_existing" {
   {{$key}} = "{{$value}}"
 		{{- end }}
   {{- end}}
+  {{- if $.datasource_required }}
```

Comment:
what is the use-case of this conditional?

Reply:
<img width="1121" alt="Screenshot 2024-07-18 at 4 09 45 PM" src="https://github.com/user-attachments/assets/3ad2f551-0ea9-41a7-8a00-b79eaa89da23">

The testConfigFvSiteAssociatedNotExisting tests throw an error if there is a depends_on because that resource isn't defined anywhere. I believe that only classes that have a datasource_required attribute need that.